src/document-converter.ts

```diff
@@ -1041,49 +1047,75 @@ export class DocumentConverter {
   }
 
   /**
+   * Converts an HTML Document's path from old world to new. Use new NPM naming
+   * as needed in the path, and change any .html extension to .js.
+   */
+  private convertDocumentUrl(htmlUrl: OriginalDocumentUrl):
+      ConvertedDocumentUrl {
+    // TODO(fks): This can be removed later if type-checking htmlUrl is enough
+    if (htmlUrl.startsWith('.') || htmlUrl.startsWith('/')) {
+      throw new Error(
+          `convertDocumentUrl() expects an OriginalDocumentUrl string` +
+          `from the analyzer, but got "${htmlUrl}"`);
+    }
+    // Use the layout-specific UrlHandler to convert the URL.
+    let jsUrl: string = this.urlHandler.convertUrl(htmlUrl);
+    // Temporary workaround for imports of some shadycss files that wrapped
+    // ES6 modules.
+    if (jsUrl.endsWith('shadycss/apply-shim.html')) {
+      jsUrl = jsUrl.replace(
+          'shadycss/apply-shim.html', 'shadycss/entrypoints/apply-shim.js');
+    }
+    if (jsUrl.endsWith('shadycss/custom-style-interface.html')) {
+      jsUrl = jsUrl.replace(
+          'shadycss/custom-style-interface.html',
+          'shadycss/entrypoints/custom-style-interface.js');
+    }
+    // Convert any ".html" URLs to point to their new ".js" module equivilent
+    jsUrl = replaceHtmlExtensionIfFound(jsUrl);
+    return jsUrl as ConvertedDocumentUrl;
+  }
+
+  /**
+   * Converts the URL for a script that is already being loaded in a
+   * pre-conversion HTML document via the <script> tag. This is similar to
+   * convertDocumentUrl(), but can skip some of the more complex .html -> .js
+   * conversion/rewriting.
+   */
+  private convertScriptUrl(oldUrl: OriginalDocumentUrl): ConvertedDocumentUrl {
+    // TODO(fks): This can be removed later if type-checking htmlUrl is enough
+    if (oldUrl.startsWith('.') || oldUrl.startsWith('/')) {
```

Comment:
Same logic as before.  Even if not using a regexp, this check should probably still be methodized with an intention-revealing name.